sycl/test/extensions/annotated_usm/annotated_usm.cpp

```diff
@@ -0,0 +1,243 @@
+  auto APtr53 = aligned_alloc_device_annotated(16, N, q);
+  static_assert(
+      std::is_same_v<
+          decltype(APtr52),
+          annotated_ptr<void, decltype(properties{usm_kind<alloc::device>})>>);
+  static_assert(std::is_same_v<decltype(APtr52), decltype(APtr53)>);
+
+  auto APtr6 = malloc_host_annotated<int>(N, q);
+  static_assert(
+      std::is_same_v<
+          decltype(APtr6),
+          annotated_ptr<int, decltype(properties{usm_kind<alloc::host>})>>);
+
+  auto APtr7 = malloc_shared_annotated<int>(N, q);
+  static_assert(
+      std::is_same_v<
+          decltype(APtr7),
+          annotated_ptr<int, decltype(properties{usm_kind<alloc::shared>})>>);
+
+  // The USM kinds differ
+  static_assert(!std::is_same_v<decltype(APtr6), decltype(APtr7)>);
+
+  free(APtr1, q);
+  free(APtr11, q);
+  free(APtr12, q);
+  free(APtr2, q);
+  free(APtr3, q);
+  free(APtr4, q);
+  free(APtr41, q);
+  free(APtr42, q);
+  free(APtr5, q);
+  free(APtr51, q);
+  free(APtr52, q);
+  free(APtr53, q);
+  free(APtr6, q);
+  free(APtr7, q);
+}
+
+void TestProperty(sycl::queue &q) {
+  // P4: input properties
+  properties P4{conduit, cache_config{large_slm}};
+  // ANNP4: output properties, with no usm_kind and no runtime property
+  properties ANNP4{conduit};
+  auto APtr1 = malloc_annotated<int>(N, q, alloc::device, P4);
+  static_assert(
+      std::is_same_v<decltype(APtr1), annotated_ptr<int, decltype(ANNP4)>>);
+
+  // APtr7 is of type annotated_ptr<int, decltype(properties{})>;
+  auto APtr2 = malloc_annotated<int>(N, q, alloc::device);
+  static_assert(std::is_same_v<decltype(APtr2), annotated_ptr<int>>);
+
+  properties P5{usm_kind<alloc::host>};
+  // Throws an exception with error code errc::invalid
+  // auto APtr8 = malloc_annotated<int>(N, q, sycl::usm::alloc::device, P5);
+
+  // Error: the USM kinds do not agree
+  // auto APtr9 = malloc_shared_annotated<int>(N, q, P5);
+
+  // the USM kind is required in the propList
+  auto APtr3 = malloc_annotated<int>(N, q, P5);
+  static_assert(
+      std::is_same_v<decltype(APtr3), annotated_ptr<int, decltype(P5)>>);
+
+  auto APtr31 = malloc_annotated(N, q, P5);
+  static_assert(
+      std::is_same_v<decltype(APtr31), annotated_ptr<void, decltype(P5)>>);
+
+  free(APtr1, q);
+  free(APtr2, q);
+  free(APtr3, q);
+  free(APtr31, q);
+}
+
+void TestAlign(sycl::queue &q) {
+  // P7: input properties
+  properties P7{alignment<512>};
+  // ANNP7: output properties with usm_kind
+  properties ANNP7{alignment<512>, usm_kind<alloc::device>};
+  // The raw pointer of APtr1 is 512-byte aligned
+  auto APtr1 = malloc_device_annotated<int>(N, q, P7);
+  static_assert(
+      std::is_same_v<decltype(APtr1), annotated_ptr<int, decltype(ANNP7)>>);
+
+  // P8: input properties
+  properties P8{alignment<2>};
+  // ANNP8: output properties with usm_kind
+  properties ANNP8{alignment<2>, usm_kind<alloc::device>};
+  // The raw pointer of APtr2 is sizeof(int)-byte aligned, e.g., 4 for some
```

Comment:
Can you add the following assert to verify that the pointer is actually 4-aligned:
assert(((uintptr_t)APtr2.get() & 3) == 0);

Similarly, for each of these test cases can you add the runtime assert to check the pointer's aligment?

Reply:
Tests in `sycl/test/extensions/...` directory only runs syntax check, so the alignment of the returned pointer cannot be verified.
The alignment verification is in the test `sycl/test-e2e/Annotated_usm/...`